src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -285,6 +361,33 @@ public function prepare_item_for_response( $post, $request ) { // phpcs:ignore V
 		return $response;
 	}
 
+	/**
+	 * Get the post, if the ID is valid.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param int $id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_post( $id ) {
+		$error = new WP_Error(
+			'rest_global_styles_not_found',
+			__( 'No global styles config exist with that id.' ),
+			array( 'status' => 404 )
+		);
+
+		if ( (int) $id <= 0 ) {
+			return $error;
+		}
+
+		$post = get_post( (int) $id );
```

Comment:
Nit-pick: we don't have to cast `$id` to an integer twice.
It's enough to only do it once.
```suggestion
                $id = (int) $id;
		if ( $id <= 0 ) {
			return $error;
		}

		$post = get_post( $id );
```

Reply:
This code is copied and pasted from post controller...

Reply:
Fixed this anyway.